src/plugins/gui/fiatPluginTypes.ts

```diff
@@ -129,3 +115,8 @@ export interface FiatPlugin {
 }
 
 export type FiatPluginFactory = (params: FiatPluginFactoryArgs) => Promise<FiatPlugin>
+
+export interface StatefulSceneEvent<EventValue, State extends object> {
```

Comment:
I won't ask for a change, but not sure what this commit buys us. The syntax for calling the callbacks is more complex. The generics also make the code harder to read.